plugins/com.google.cloud.tools.eclipse.dataflow.core/src/com/google/cloud/tools/eclipse/dataflow/core/project/GcsDataflowProjectClient.java

```diff
@@ -88,25 +90,30 @@ public StagingLocationVerificationResult createStagingLocation(
     }
   }
 
-  private static String toGcsBucketName(String stagingLocation) {
-    String gcsLocation;
-    if (stagingLocation.startsWith(GCS_PREFIX)) {
-      gcsLocation = stagingLocation.substring(GCS_PREFIX.length());
-    } else {
-      gcsLocation = stagingLocation;
-    }
-    String bucketName;
-    if (gcsLocation.indexOf('/') < 0) {
-      bucketName = gcsLocation;
+  /**
+   * Extracts a bucket name from a given GCS URL. For example, {@code "/bucket/object"} returns
```

Comment:
Worth mentioning that it assumes the input is either a valid GCS URL or of the form `bucket/path` or `/bucket/path`.  Results may be unpredictable if not a valid location.